pkg/network/ebpf/c/protocols/tls/sowatcher-types.h

```diff
@@ -0,0 +1,34 @@
+#ifndef __SOWATCHER_TYPES_H
+#define __SOWATCHER_TYPES_H
+
+#include "ktypes.h"
+
+#define SO_SUFFIX_SIZE 3
+#define LIB_PATH_MAX_SIZE 120
+
+typedef struct {
+    __u32 pid;
+    __u32 len;
+    char buf[LIB_PATH_MAX_SIZE];
+} lib_path_t;
+
+typedef struct {
+    unsigned short common_type;
+    unsigned char common_flags;
+    unsigned char common_preempt_count;
+    int common_pid;
+    long syscall_nr;
+} trace_common;
```

Comment:
I think using a separate struct can cause padding to be inserted at the end, thus potentially causing alignment issues. I see that CWS always uses a single struct and doesn't embed anything.